wordpress-mobile/WordPress-Android
# Refresh BP card when date changes

This PR fixes and issue where Blogging Prompt card is not updated after Day changes and app is not restarted.

Gist of the problem is that to update the card we setup flow and keep continuously collecting from it. Problem is that the flow is setup only once, using set date, and even if we refresh the data, or the day changes the flow still collects from previous day.

As a workaround I collect all the prompts, and filter them in the flow. This way we always get fresh prompt to display.

To test:
- Launch JP app and confirm that BP card is visible.
- Answer prompt and confirm that the card updates.
- While the app is open, navigate to device settings and change clock to the next day.
- Return to the app, and confirm that dashboard is refreshing, and different prompt is visible.


## Regression Notes
1. Potential unintended areas of impact
- Potential unexpected issue with prompts.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Updated existing tests.

PR submission checklist:
